Drv/TcpServer/test/ut/Tester.cpp

```diff
@@ -0,0 +1,209 @@
+#include <Drv/Ip/test/ut/PortSelector.hpp>
+#include <Drv/Ip/test/ut/SocketTestHelper.hpp>
+
+Os::Log logger;
+
+#define INSTANCE 0
+#define MAX_HISTORY_SIZE 1000
+
+
+namespace Drv {
+
+// ----------------------------------------------------------------------
+// Construction and destruction
+// ----------------------------------------------------------------------
+
+void Tester ::test_with_loop(U32 iterations, bool recv_thread) {
+    U8 buffer[sizeof(m_data_storage)];
+    Drv::SocketIpStatus status1 = Drv::SOCK_SUCCESS;
+    Drv::SocketIpStatus status2 = Drv::SOCK_SUCCESS;
+    Drv::SocketIpStatus serverStat = Drv::SOCK_SUCCESS;
+
+    U16 port =  Drv::Test::get_free_port();
+    ASSERT_NE(0, port);
+
+    this->component.configure("127.0.0.1", port, 0, 100);
+    status2 = this->component.startup();
+    EXPECT_EQ(serverStat, SOCK_SUCCESS);
+
+    // Start up a receive thread
+    if (recv_thread) {
+        Fw::EightyCharString name("receiver thread");
+        this->component.startSocketTask(name, 50, 1024, true);
+    }
+
+    // Loop through a bunch of client disconnects
+    for (U32 i = 0; i < iterations && serverStat == SOCK_SUCCESS; i++) {
+        Drv::TcpClientSocket client;
+        client.configure("127.0.0.1", port, 0, 100);
+        status2 = client.open();
+
+        I32 size = sizeof(m_data_storage);
+
+        // Not testing with reconnect thread, we will need to open ourselves
+        if (not recv_thread) {
+            status1 = this->component.open();
+        } else {
+            EXPECT_TRUE(Drv::Test::wait_on_change(this->component.getSocketHandler(), true, SOCKET_RETRY_INTERVAL_MS/10 + 1));
+        }
+        EXPECT_TRUE(this->component.getSocketHandler().isOpened());
+
+        EXPECT_EQ(status1, Drv::SOCK_SUCCESS);
+        EXPECT_EQ(status2, Drv::SOCK_SUCCESS);
+
+        // If all the opens worked, then run this
+        if (Drv::SOCK_SUCCESS == status1 && Drv::SOCK_SUCCESS == status2 &&
+            this->component.getSocketHandler().isOpened()) {
+            // Force the sockets not to hang, if at all possible
+            Drv::Test::force_recv_timeout(this->component.getSocketHandler());
+            Drv::Test::force_recv_timeout(client);
+            m_data_buffer.setSize(sizeof(m_data_storage));
+            Drv::Test::fill_random_buffer(m_data_buffer);
+            Drv::SendStatus status = invoke_to_send(0, m_data_buffer);
+            EXPECT_EQ(status, Drv::SEND_OK);
+            status2 = client.recv(buffer, size);
+            EXPECT_EQ(status2, Drv::SOCK_SUCCESS);
+            EXPECT_EQ(size, m_data_buffer.getSize());
+            Drv::Test::validate_random_buffer(m_data_buffer, buffer);
+
+            // If receive thread is live, try the other way
+            if (recv_thread) {
+                m_spinner = false;
+                m_data_buffer.setSize(sizeof(m_data_storage));
+                client.send(m_data_buffer.getData(), m_data_buffer.getSize());
+                while (not m_spinner) {}
```

Comment:
I the empty while loop body intentional here?

Reply:
Yes, it just loops until the variable in another thread flips.